samples/matter/thermostat/README.rst

```diff
@@ -0,0 +1,352 @@
+
+The ``build_nrf52840dk_nrf52840`` parameter specifies the output directory for the build files.
+
+.. note::
+   If the selected board does not support the selected build type, the build is interrupted.
+   For example, if the ``shell`` build type is not supported by the selected board, the following notification appears:
+
+   .. code-block:: console
+
+      File not found: ./ncs/nrf/samples/matter/thermostat/configuration/nrf52840dk_nrf52840/prj_shell.conf
+
+.. _matter_thermostat_testing:
+
+Testing
+=======
+
+After building the sample and programming it to your development kit, you can either test the sample's basic features or use the Matter weather station application to :ref:`test the thermostat with an external sensor <matter_thermostat_sensor_testing>`.
+
+.. _matter_thermostat_sample_basic_features_tests:
+
+Testing basic features
+----------------------
+
+After building the sample and programming it to your development kit, complete the following steps to test its basic features:
+
+1. |connect_kit|
+#. |connect_terminal_ANSI|
+#. Observe that the **LED1** starts flashing (short flash on).
+   This means that the sample has automatically started the Bluetooth LE advertising.
+#. Observe the UART terminal.
+   The sample starts automatically printing the simulated temperature data to the terminal with 30-second intervals.
+#. Press **Button 2** to print the thermostat temperature data to the terminal.
+#. Press **Button 1** for six seconds to initiate the factory reset of the device.
+
+The device reboots after all its settings are erased.
+
+.. _matter_thermostat_sensor_testing:
+
+Testing with external sensor
+----------------------------
+
+After building this sample and the :ref:`Matter weather station <matter_weather_station_app>` application and programming them to the development kits, complete the following steps to test communication between both devices:
+
+1. |connect_kit|
+#. |connect_terminal_ANSI|
+#. If devices were not erased during the programming, press the button responsible for the factory reset on each device.
+#. On each device, press the button that starts the Bluetooth LE advertising.
+#. Commission devices to the Matter network.
+   See `Commissioning the device`_ for more information.
+   During the commissioning process, write down the values for the thermostat node ID, the temperature sensor node ID, and the temperature sensor endpoint ID.
+   These IDs are going to be used in the next steps (*<thermostat_node_ID>*, *<temperature_sensor_node_ID>*, and *<thermostat_sensor_endpoint_ID>*, respectively).
+#. Use the :doc:`CHIP Tool <matter:chip_tool_guide>` ("Writing ACL to the ``accesscontrol`` cluster" section) to add proper ACL for the temperature sensor device.
+   Use the following command, with *<thermostat_node_ID>* and *<temperature_sensor_endpoint_ID>* values from the previous step about commissioning:
```

Comment:
Maybe these template names should directly reflect the ones used in the command template? I mean e.g. `<thermostat_node_ID>` vs. `<Thermostat NodeId>`

Reply:
Can we edit the command template? Every other piece of Matter docs uses underscored variables.